controlpanel/frontend/static/components/selectable-rows/selectable-rows.js

```diff
@@ -1,24 +1,92 @@
+/**
+ * Module to allow selecting entire rows in a table instead of using a small checkbox.
+ *
+ * It will disable any in-row actions buttons once a selection is made, but will re-enable them once
+ * all the selections are cleared.
+ *
+ * It can optionally support enabling a (global)
+ * button(s) that lives outside the table.
+ * How to use:
+ *  1. Add the class "selectable-rows" to your <table>
+ *  2. For any button you want to enable when there are selections add the class "selectable-rows__enable-on-selections"
+ *  3. Use the macro `row_selector` to put insert an invisible input box in the first column of your table
+ * The module will automatically disable any action buttons that are in side the row.
+ *
+ * WARNING: This module doesn't support more than 1 table on the same page
+ * @type {{init(): void, onSelectionChanged(*): void, baseClass: string, bindEvents(): void}}
+ */
+
 moj.Modules.selectable_rows = {
-  selector: '.selectable-rows tr',
+  baseClass: "selectable-rows",
 
   init() {
-    if (document.querySelector(this.selector)) {
+    this.tableRowSelector = `.${this.baseClass} tr`;
+    this.tableRowButtonSelector = `${this.tableRowSelector} button`;
+    this.selectionClass = `${this.baseClass}__row-selected`;
+    this.enableOnSelectionSelector = `.${this.baseClass}__enable-on-selections`;
+    this.selectionChangedEventName = `${this.baseClass}:rowSelectionChanged`;
+    if (document.querySelector(this.tableRowSelector)) {
       this.bindEvents();
     }
   },
 
+  /**
+   * Runs when rows are selected/deselected
+   * @param event
+   */
+  onSelectionChanged(event) {
+    const rowButtons = document.querySelectorAll(this.tableRowButtonSelector);
+    const enableOnSelectionButtons = document.querySelectorAll(
+      this.enableOnSelectionSelector
+    );
+    if (event.detail.rowsSelected.length > 0) {
+      // some rows are selected
+      rowButtons.forEach(e => (e.disabled = true));
+      enableOnSelectionButtons.forEach(e => e.disabled = false);
+    } else {
+      rowButtons.forEach(e => (e.disabled = false));
+      enableOnSelectionButtons.forEach(e => e.disabled = true);
+    }
+  },
+
   bindEvents() {
-    document.querySelectorAll(this.selector).forEach(row => {
+    window.addEventListener(this.selectionChangedEventName, e =>
+      this.onSelectionChanged(e)
+    );
+    document.querySelectorAll(this.tableRowSelector).forEach(row => {
```

Comment:
Probably not or not meaningful anyway but is there any performance hit by creating a bunch of anonymous functions in a loop here? Possibly fascinating but not important.

A more appropriate question would be, any reason to not have this as a named function (something like `onRowClicked()`)? Like we have `onSelectionChanged()`.

Just curious, nothing wrong with this of course :)